linkml/generators/pythongen.py

```diff
@@ -836,7 +930,16 @@ def gen_postinit(self, cls: ClassDefinition, slot: SlotDefinition) -> Optional[s
             ):
                 rlines.append(f"\tself.{aliased_slot_name} = {base_type_name}()")
             else:
-                if (
+                if slot.range in self.schema.enums and slot.ifabsent:
+                    # `ifabsent` for an enumeration cannot be assigned to
+                    # the dataclass field default, because it would be a
+                    # mutable. `python_ifabsent_processor.py` can specify
+                    # the default as string and here that string gets
+                    # converted into an object attribute invocation
+                    rlines.append(f"\tenum_name, pv_name = self.{aliased_slot_name}.split('.')")
+                    rlines.append("\tenum_cls = globals().get(enum_name)")
+                    rlines.append(f"\tself.{aliased_slot_name} = getattr(enum_cls, pv_name)")
```

Comment:
OK i like the string approach, and as usual am truly not trying to be a nitpicker here, but here's an example that fails because names on enums aren't sanitized:

```yaml
CordialnessEnum:
  permissible_values:
    # ...
    ch.aotic:
      description: doesn't obey conventions

# ...

IfAbsent:
  attributes:
    ifabsent_dotted_literal:
      range: CordialnessEnum
      ifabsent: CordialnessEnum(ch.aotic)
  
```

Then we get 

```python
@dataclass(repr=False)
class IfAbsent(YAMLRoot):
    # ....
    ifabsent_dotted_literal: Optional[Union[str, "CordialnessEnum"]] = 'CordialnessEnum.ch.aotic'

    def __post_init__(self):
        # ...
        if self.ifabsent_dotted_literal is not None and not isinstance(self.ifabsent_dotted_literal, CordialnessEnum):
            enum_name, pv_name = self.ifabsent_dotted_literal.split('.')
            enum_cls = globals().get(enum_name)
            self.ifabsent_dotted_literal = getattr(enum_cls, pv_name)
```

which fails with the relatively uninformative error message: 

```
(Pdb) ksm.IfAbsent()
*** ValueError: too many values to unpack (expected 2)
```

I think that the easiest thing to do here would be to mimic the behavior of [`StrEnum`](https://docs.python.org/3/library/enum.html#enum.StrEnum) and just put the name of the enum value rather than the Enum and the value

So then that simplifies to 


```python
@dataclass(repr=False)
class IfAbsent(YAMLRoot):
    # ....
    ifabsent_dotted_literal: Optional[Union[str, "CordialnessEnum"]] = 'ch.aotic'

    def __post_init__(self):
        # ...
        if self.ifabsent_dotted_literal is not None and not isinstance(self.ifabsent_dotted_literal, CordialnessEnum):
            self.ifabsent_dotted_literal = CordialnessEnum(self.ifabsent_dotted_literal)
```

which, remarkably, works.

```python
(Pdb) ksm.CordialnessEnum('ch.aotic')
CordialnessEnum(text='ch.aotic', description="doesn't obey convention")
```

Reply:
Good catch! You are fully right, we do not need the enum name, since the range already constraints it. So only passing the permissible value suffice.

Reply:
BTW ensuring that corner cases are properly covered is IMO not nitpicking, but high quality code 😉